### PR TITLE
Arrange distributions and sub-contents alphabetically

### PR DESCRIPTION
**Pull request by Dhruvanshu-Joshi**
**What is this PR about?**
Aimed to solve issue [6621](https://github.com/pymc-devs/pymc/issues/6621).Updated the api docs for distributions and its sub-content to arrange the listed distributions in alphabetical order.

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [ ] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## New features
- ...

## Bugfixes
- ...

## Documentation
- Did build sphinx locally and got the desired output. ```Distributions utilities``` is intentionally kept at the bottom in ```distributions.rtx```.

## Maintenance
- ...

----
:books: Documentation preview :books:: https://pymc--6653.org.readthedocs.build/en/6653/

### CODE REVIEW

**Comment by ricardoV94** [comment]
I am not sure we want alphabetically sorting at all levels... There's a question of what we prioritize as well

**Comment by Dhruvanshu-Joshi** [comment]
Hey @ricardoV94 . Based on the discussion in issue [6621](https://github.com/pymc-devs/pymc/issues/6621), I felt that sorting the distributions in alphabetical order is the next sensible step. How would you like me to change the order of the listed distributions?

**Comment by ricardoV94** [comment]
I think the distributions is sensible but not necessarily the order of the categories (continuous vs discrete vs multivariate, etc). Others may disagree with me though.

**Comment by reshamas** [comment]
> I think the distributions is sensible but not necessarily the order of the categories (continuous vs discrete vs multivariate, etc). Others may disagree with me though.

- I agree that the categories don't need to be in alphabetical order. 
- "Types of distributions" could be alphabetical. Continuous and discrete are the most widely referred (I assume). Alphabetical puts these at or near the top anyway: "Continuous" and "Categorical"
- Within "Continuous", if those distributions were in alphabetical order, it would be much easier to find, especially since there are 30+ types of continuous distributions:
   - Beta
   - Cauchy
   - etc. 
- Same for "Discrete".

**Comment by Dhruvanshu-Joshi** [comment]
Thank You @ricardoV94 @reshamas . Just so that I get this right this time, I'll summarize the changes to be made that I understand. Firstly, keep categories( i.e. continuous vs discrete vs multivariate, etc) in the same order as is currently displayed on the website. Then, alphabetically arrange the types of distributions under each category( eg: continuous, discrete, etc).

**Review by ricardoV94 — APPROVED** [review]

**Review by larryshamalama — APPROVED** [review]
Looks good, thanks @Dhruvanshu-Joshi 🙂 I also notice that you added the distributions that were forgotten in the documentation, which is much appreciated!

File: docs/source/api/distributions/continuous.rst [hunk 1]
```diff
@@ -6,36 +6,36 @@ Continuous
...
+   HalfFlat
+   HalfNormal
+   HalfStudentT
+   Interpolated
+   InverseGamma
+   Kumaraswamy
+   Laplace
    Logistic
+   LogNormal
    LogitNormal
```
Comment (reshamas) [inline docs/source/api/distributions/continuous.rst]:
```suggestion
   Laplace
   Logistic
   LogitNormal
   LogNormal
```

**Comment by Dhruvanshu-Joshi** [comment]
> Looks good, thanks @Dhruvanshu-Joshi 🙂 I also notice that you added the distributions that were forgotten in the documentation, which is much appreciated!

Thank You @larryshamalama ☺️

**Comment by Dhruvanshu-Joshi** [comment]
Hey @reshamas , I have included the suggestion in the latest commit. Hope it solves the issue.

File: docs/source/api/distributions/continuous.rst [hunk 2]
```diff
@@ -6,36 +6,36 @@ Continuous
...
    Moyal
-   AsymmetricLaplace
+   Normal
+   Pareto
    PolyaGamma
+   Rice
+   SkewNormal
+   StudentT
+   TruncatedNormal
+   Triangular
```
Comment (reshamas) [inline docs/source/api/distributions/continuous.rst]:
```suggestion
   Triangular
   TruncatedNormal
```

File: docs/source/api/distributions/mixture.rst [hunk 3]
```diff
@@ -6,5 +6,5 @@ Mixture
 .. autosummary::
    :toctree: generated
 
-   Mixture
    NormalMixture
+   Mixture
```
Comment (reshamas) [inline docs/source/api/distributions/mixture.rst]: Shouldn't "Mixture" come before "NormalMixture"?

File: docs/source/api/distributions/utilities.rst [hunk 4]
```diff
@@ -6,8 +6,8 @@ Distribution utilities
 .. autosummary::
    :toctree: generated/
 
-    Distribution
-    Discrete
     Continuous
     CustomDist
+    Distribution
+    Discrete
```
Comment (reshamas) [inline docs/source/api/distributions/utilities.rst]:
```suggestion
    Discrete
    Distribution
```

**Comment by reshamas** [comment]
Thank you @Dhruvanshu-Joshi.
I have no more comments. Looks great.

**Comment by reshamas** [comment]
@ricardoV94 
This is a doc fix, so it seems not all the checks are running.

cc:  @OriolAbril

**Comment by ricardoV94** [comment]
We didn't change the required yet to not force old PRs to update.

**Pull request merged**